plugins/discourse-local-dates/assets/javascripts/discourse/components/discourse-local-dates-create-form.js.es6

```diff
@@ -15,23 +15,154 @@ export default Ember.Component.extend({
   recurring: null,
   advancedMode: false,
   isValid: true,
+  timezone: null,
+  timezones: null,
 
   init() {
-    this._super();
+    this._super(...arguments);
 
-    this.set("date", moment().format(this.dateFormat));
-    this.set("timezones", []);
-    this.set(
-      "formats",
-      (this.siteSettings.discourse_local_dates_default_formats || "")
+    this.setProperties({
+      timezones: [],
+      formats: (this.siteSettings.discourse_local_dates_default_formats || "")
         .split("|")
-        .filter(f => f)
-    );
+        .filter(f => f),
+      timezone: moment.tz.guess(),
+      date: moment().format(this.dateFormat)
+    });
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    this._renderPreview();
+  },
+
+  _renderPreview: debounce(function() {
+    const markup = this.get("markup");
+
+    if (markup) {
+      cookAsync(markup).then(result => {
+        this.set("currentPreview", result);
+
+        Ember.run.next(() =>
```

Comment:
I always dislike this kind nested `Ember.run` - is the issue that when the renderPreview is triggered the element it renders into might not exist yet?

Reply:
I think this is not needed anymore, it's the result of a refactoring where I have removed something in between